### ISSUE TITLE
Switching from AVA to Jest

### ISSUE THREAD

**Issue by skellock** [infinitered/ignite-ir-next#10]
There's been a lot of talk about this [here](https://github.com/infinitered/ignite/issues/781), in a few other tickets and now in Slack.

Let's switch to Jest for our React Native projects (even though I love AVA and will still use it everywhere else -- including any ignite plugins I write).  

React Native seems to have standardized on Jest and seems a bit against the grain to stay with AVA here.

Can we get consensus on this?  We've got some contributors ready to mobilize and we need to make this decision.

/cc @jamonholmgren @GantMan

**Comment by jamonholmgren** [comment]
Added to our next Ignite core team meeting agenda.

**Comment by jamonholmgren** [comment]
cc @derekgreenberg @robinheinze

**Comment by derekgreenberg** [comment]
I used [jest-codemods](https://github.com/skovhus/jest-codemods) to port the Ava unit tests over to Jest for a project (built from ignite 2.0) and it worked perfectly.

**Comment by skellock** [comment]
Sorceror!

**Comment by derekgreenberg** [comment]
Ok, might as well show some other changes made for this:

Added /jest directory

In package.json under 'scripts:

Replace Ava stuff with Jest stuff:

Remove
```
    "test": "ava",
     "test:watch": "ava --watch",
     "coverage": "nyc ava && nyc report --reporter=html && open coverage/index.html || xdg-open coverage/index.html",
```
Replace:

```
    "test": "jest",
    "updateSnapshot": "jest --updateSnapshot",
    "test:watch": "jest --watch",
    "coverage": "nyc jest && nyc report --reporter=html && open coverage/index.html || xdg-open coverage/index.html",
```

` in package.json, removed Ava node `
```
 "ava": {
     "files": [
       "Tests/**/*.js",
       "!Tests/Setup.js"
     ],	
     "require": [		
       "babel-register",		
       "babel-polyfill",		
       "react-native-mock/mock",		
      "./Tests/Setup"		
    ],		
    "babel": "inherit"
```
`in package.json added Jest node`

```
  "jest": {
    "preset": "react-native",
    "setupFiles": [
      "./jest/setup.js"
    ]
  },
```

`Removed unused devDependencies from package.json`

```
"enzyme": "^2.6.0",
"react-dom": "^15.4.0",
"react-native-mock": "^0.3.1",
```

`Added Jest devDependencies in package.json`

"babel-preset-react-native": "1.9.1",
"jest": "^19.0.2",
 "jest-react-native": "^18.0.0",
 "react-test-renderer": "^15.5.4",

Added: jest/setup.js

// add any mocks you need, based on any mocks you created previously for Ava. The following is 
// just an example that you can use to create your own mock!!!

jest
.mock('react-native-device-info', () => {
  return { isTablet: jest.fn() }
})
.mock('react-native-router-flux', () => {
  return {
    Actions: {'myScreen': () => {}},
    ActionConst: {RESET: 'reset'}
  }
})
.mock('react-native-keyboard-spacer', () => 'KeyboardSpacer')
.mock('react-native-i18n', () => {
  const english = require('../App/I18n/languages/english.json')
  const keys = require('ramda')
  const replace = require('ramda')
  const forEach = require('ramda')

  return {
    t: (key, replacements) => {
      let value = english[key]
      if (!value) return key
      if (!replacements) return value

      forEach((r) => {
        value = replace(`{{${r}}}`, replacements[r], value)
      }, keys(replacements))
      return value
    }
  }
})

**Comment by jamonholmgren** [comment]
Fixed in #30 . :tada: :taco:
